core/context.go

```diff
@@ -26,17 +26,30 @@ import (
 	"go.uber.org/fx/service"
 )
 
-// Context embedds Host and go context for use
+// Context embeds Host and go context for use
 type Context interface {
 	gcontext.Context
 	service.Host
+	WithContext(ctx gcontext.Context) *context
```

Comment:
why are we import aliasing gocontext? it doesn't clash? we should just import it as `context` and use `context.Context` here unless I'm missing something